internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,99 @@
+package telemetry
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/graph"
+)
+
+type DataCollector interface {
+	Collect(ctx context.Context) Data
+}
+
+type GraphGetter interface {
+	GetLatestGraph() *graph.Graph
+}
+
+type GraphResourceCount struct {
+	Gateways       int
+	GatewayClasses int
+	HTTPRoutes     int
+	Secrets        int
+	Services       int
+}
+
+type ProjectNameAndVersion struct {
```

Comment:
nit: it's best to avoid names that include `and`. What about:

```suggestion
type ProjectMetadata struct {
```